### PR TITLE
coprv2: plugin version checks

### PR DESCRIPTION
**Pull request by Jibbow**
### What problem does this PR solve?

Introduces version checks for API version and rustc version of plugins.

Issue Number: #9747 

Problem Summary:

When loading a coprocessor plugin, we need to make sure that it actually runs. 
Thus, we need to make sure that the following things match up:
 * the version of the `coprocessor_plugin_api` crate the plugin was compiled with.
 * the rustc version the plugin was compiled with (has to be the same as the rustc version TiKV was compiled with because Rust does not guarantee a stable ABI)
 * the `target` platform the plugin was compiled for and the target platform of TiKV

### What is changed and how it works?

All in all, it is very similar to the [POC](https://github.com/andylokandy/plugin).

Proposal: [Coprocessor V2 RFC](https://github.com/andylokandy/rfcs/blob/plugin/text/2021-02-24-coprocessor-plugin.md) 

What's Changed:

### Related changes

- based on PR #9925 

### Check List 

Tests 

- Unit test (TODO: I'm not entirely sure how to test this, help appreciated!)

### Release note 

 * No release note.

### CODE REVIEW

**Comment by Jibbow** [comment]
Not entirely sure why the build fails. Any ideas?

**Comment by skyzh** [comment]
Seems that rustc ICE... Is there any way to clear cache on CI? @andylokandy

**Comment by skyzh** [comment]
Also I saw `<jemalloc>: Invalid conf pair: prof:true`. Seems to be caused by jemalloc?

**Comment by andylokandy** [comment]
Cleaning the cache should work...

**Comment by Jibbow** [comment]
Nice, thanks!

File: src/coprocessor_v2/plugin_registry.rs [hunk 1]
```diff
@@ -10,6 +10,51 @@ use std::sync::{mpsc, Arc, RwLock};
 use std::thread;
 use std::time::Duration;
 
+quick_error! {
```
Comment (andylokandy) [inline src/coprocessor_v2/plugin_registry.rs:13]: Prefer to use `anyhow` and `thiserror`.

**Pull request closed without merging**